R/target_market_share.R

```diff
@@ -83,10 +83,14 @@ target_market_share <- function(data,
   )
 
   if (by_company & weight_production) {
-    warn(message = "You have selected to view the weighted production at company
-         level. Make sure this is what you want to see! It is often desirable to
-         see these results unweighted, to understand where the company actualyl
-         stands in the market. ")
+    warn(
+      class = "incompatible_by_company_and_weight_production",
+      glue(
+        "`by_company` & `weight_production` shouldn't be both TRUE or FALSE`:
+        You've supplied `by_company = TRUE` and `weight_production = TRUE`.
+        Did you forget to set one of these arguments to `FALSE`?"
```

Comment:
This is a bit too strict of a warning for me. They shouldn't both be TRUE unless you are sure you want them to be, and that has to do with the context of what you want to analyze...